File: my-folder/0102-binary-tree-level-order-traversal/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if (root == null) return [];
    let queue = [root];
    let result = [];
    while (queue.length > 0) {
        let currLength = queue.length;
        let currLevel = [];
        for (let i = 0; i < currLength; i++) {
            if (queue[0].left != null) {
                queue.push(queue[0].left);
            }
            if (queue[0].right != null) {
                queue.push(queue[0].right);
            }
            currLevel.push(queue[0].val);
            queue.shift();
        }
        result.push(currLevel);
    }

    return result;
};

File: my-folder/0102-binary-tree-level-order-traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if (root == null) return List.of();
        Queue<TreeNode> q = new ArrayDeque<>();
        
        List<List<Integer>> result = new ArrayList<>();
        q.offer(root);
        while (!q.isEmpty()) {
            int levelSize = q.size();
            List<Integer> level = new ArrayList<>();
            while (levelSize > 0) {
                TreeNode curr = q.poll();
                level.add(curr.val);
                
                if (curr.left != null) q.offer(curr.left);
                if (curr.right != null) q.offer(curr.right);
                levelSize--;
            }
            
            result.add(level);
        }
        return result;
    }
}
